storage/interceptors/smaap_virtual_resource_cascade_operation_create_interceptor.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package interceptors
+
+import (
+	"context"
+	"github.com/Peripli/service-manager/operations"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/storage"
+	"time"
+)
+
+const VirtualResourceCascadeOperationCreateInterceptorProviderName = "VirtualResourceCascadeOperationCreateInterceptorProvider"
+
+type VirtualResourceCascadeOperationCreateInterceptor struct {
+	TenantIdentifier string
+}
+
+type VirtualResourceCascadeOperationCreateInterceptorProvider struct {
+	TenantIdentifier string
+}
+
+func (c *VirtualResourceCascadeOperationCreateInterceptorProvider) Provide() storage.CreateOnTxInterceptor {
+	return &VirtualResourceCascadeOperationCreateInterceptor{
+		TenantIdentifier: c.TenantIdentifier,
+	}
+}
+
+func (c *VirtualResourceCascadeOperationCreateInterceptorProvider) Name() string {
+	return VirtualResourceCascadeOperationCreateInterceptorProviderName
+}
+
+func (co *VirtualResourceCascadeOperationCreateInterceptor) OnTxCreate(f storage.InterceptCreateOnTxFunc) storage.InterceptCreateOnTxFunc {
+	return func(ctx context.Context, storage storage.Repository, obj types.Object) (types.Object, error) {
+		operation := obj.(*types.Operation)
+		// currently we have only one virtual object: tenant
+		isVirtual := types.IsVirtualType(operation.ResourceType)
+		if !isVirtual || operation.CascadeRootID == "" || operation.Type != types.DELETE {
+			return f(ctx, storage, operation)
+		}
+
+		// init operation properties
+		operation.PlatformID = types.SMPlatform
```

Comment:
maybe init `operations` here with the relevant fields instead of creating it before you know if you need to use it?

Reply:
Done, the operation check was moved upper